dapp/src/executeTask.js

```diff
@@ -19,12 +19,75 @@ async function writeTaskOutput(path, message) {
   }
 }
 
+async function processProtectedData(
+  index,
+  {
+    IEXEC_IN,
+    IEXEC_OUT,
+    appDeveloperSecret,
+    requesterSecret,
+    datasetFilename = null,
```

Comment:
Could we compute datasetFilename using the index param rather than passing it as input? It would prevent inconsistencies.